From bb83ee0b2999cd3e7b7f4eb0adf59f182f792f11 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Tue, 8 Aug 2017 16:06:35 +0300
Subject: [PATCH 039/706] enetc: Fix Rx stall after device close

RBPIR needs to be explicitly reset when the Rx ring is re-initialized
(i.e. device close() followed by device open()).

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 3c447e4bbfe2aef7b8b8b0a6d67ed39b60fb9e3d)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc.c b/drivers/net/ethernet/freescale/enetc/enetc.c
index f5d591951c20..74d5ab2cab65 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc.c
@@ -688,6 +688,7 @@ static void enetc_setup_rx_ring(struct enetc_ndev_priv *priv)
 	/* enable ring */
 	enetc_rxbdr_wr(hw, 0, ENETC_RBMR, rbmr);
 
+	enetc_rxbdr_wr(hw, 0, ENETC_RBPIR, 0);
 	rx_ring->rcir = hw->reg + ENETC_BDR(RX, 0, ENETC_RBCIR);
 	enetc_refill_rx_ring(rx_ring, enetc_bd_unused(rx_ring));
 }
-- 
2.17.1

